cognitedata/reveal
# feat: 360 image annotation styling

#### Type of change

![Feat](https://img.shields.io/badge/Type-Feat-green)  

#### Jira ticket :blue_book:


https://cognitedata.atlassian.net/browse/REV-876

## Description :pencil:


Adds simple styling capabilities to annotations. It sort-of mimics cad and point cloud models, where the `Image360Collection` takes the place of the model. 

A big difference is that styling filters is a predicate function, rather than a "filter" object. This is mostly due to the sheer amount of data in the `AnnotationModel` object describing the annotation which may be relevant, so creating a flexible, exhaustive API based on "filter objects" may be hard. There will also be a much smaller performance hit than what we had encountered using predicate functions for node / point cloud object styling, since we only need to consider the annotations associated with the current 360 revision when applying styles.

This design is up for discussion, however.

## How has this been tested? :mag:



- Start Reveal examples
- Open https://localhost:3000/?project=3d-test&env=greenfield-3d&modelId=4923307033982497&revisionId=3100284276970593
- Load 360 image set `helideck-site-2-jpeg`
- Enter the lower-most 360-entity (has the most annotations)
- Observe that some annotations are differently colored, according to lines 178-182 in `Image360UI`
- Observe that clicking an annotation changes its color so that it feels "selected"
- Observe that selection changes when clicking new annotation
- Observe that adding a style using the new "360 annotation styling" UI colors/hides the annotations matching the label text given (the console lists the label of clicked annotations).


## Test instructions :information_source:




## Checklist :ballot_box_with_check:
